darwin/dataset/upload_manager.py

```diff
@@ -44,6 +44,35 @@ def full_path(self):
         return construct_full_path(self.data["path"], self.data["filename"])
 
 
+class FileMonitor(object):
+    """
+    An object used to monitor the progress of a :class:`BufferedReader`.
+
+    To use this monitor, you construct your :class:`BufferedReader` as you
+    normally would, then construct this object with it as argument.
+    """
+
+    def __init__(self, io: BinaryIO, file_size: int, callback):
```

Comment:
any chance we could add a type annotation to `callback`?

Reply:
That weirdly required `__future__` 🤷 
https://github.com/v7labs/darwin-py/pull/200/commits/b362f33d402e24131fef5c7d54640a92ac64909e

Reply:
Oh, this works only on python 3.7+... @andreaazzini Do you have other ideas?